### PR TITLE
Cleaned up spacing and added MARKs/ comments

### PR DESCRIPTION
**Pull request by JoshuaBayless**

### CODE REVIEW

File: Marquee/Cells/Table Cell/TableViewCell.swift [hunk 1]
```diff
@@ -8,18 +8,15 @@
...
 
 class TableViewCell: UITableViewCell, UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
     
     @IBOutlet weak var collectionView: UICollectionView!
-    
     static let identifier = "TableViewCell"
-    
-    var delegate: collectionViewDelegate?
-    
+    var delegate: CollectionViewDelegate?
```
Comment (cordhernandez) [inline Marquee/Cells/Table Cell/TableViewCell.swift:19]: It looks like the `CollectionViewDelegate` protocol is conforming to `AnyObject`, is the intention for that to make your `var delegate` weak?

File: Marquee/Models/Movies.swift [hunk 2]
```diff
@@ -12,10 +12,10 @@ struct Movies: Decodable {
...
-    var original_title: String
+    var originalTitle: String
     var overview: String
-    var poster_path: String?
-    var release_date: String
+    var posterPath: String?
+    var releaseDate: String
     var title: String
-    var vote_average: Double
+    var voteAverage: Double
```
Comment (cordhernandez) [inline Marquee/Models/Movies.swift]: Does the JSON response from the backend return properties with underscores? If so, you might want to consider using coding keys so you can properly decode all the data.

```
enum CodingKeys: String, CodingKey {
 case originalTitle = "original_Title"
}
```

File: Marquee/Models/Search.swift [hunk 3]
```diff
@@ -14,9 +14,9 @@ struct SearchResults: Decodable {
...
     var overview: String?
-    var poster_path: String?
-    var vote_average: Double?
-    var release_date: String?
+    var posterPath: String?
+    var voteAverage: Double?
+    var releaseDate: String?
     var title: String?
-    var first_air_date: String?
+    var firstAirDate: String?
```
Comment (cordhernandez) [inline Marquee/Models/Search.swift:21]: You might want to consider using coding keys so you can properly decode all the data.

File: Marquee/Models/TV.swift [hunk 4]
```diff
@@ -13,9 +13,9 @@ struct TVShows: Decodable {
...
 
 struct TVdata: Decodable {
-    var first_air_date: String
+    var firstAirDate: String
     var name: String
     var overview: String
-    var poster_path: String?
-    var vote_average: Double
+    var posterPath: String?
+    var voteAverage: Double
```
Comment (cordhernandez) [inline Marquee/Models/TV.swift]: You might want to consider using coding keys so you can properly decode all the data.

File: Marquee/Networking/JSONManager.swift [hunk 5]
```diff
@@ -54,11 +53,10 @@ struct JSONManager {
     
     func parseJSON<T:Decodable>(_ movieData: Data) -> T? {
         let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
```
Comment (cordhernandez) [inline Marquee/Networking/JSONManager.swift:57]: Ahh I see you're converting from snake case, well done! This should work and you can ignore my previous comments about coding keys. FYI you might be able to write some good unit tests around this.

File: Marquee/ViewControllers/DetailViewController.swift [hunk 6]
```diff
@@ -19,22 +19,19 @@ class DetailViewController: UIViewController {
...
-        scoreBar.progress = Float(selectedContent.vote_average*0.1)
+        userScore.text = "Rating: \(selectedContent.voteAverage)"
+        scoreBar.progress = Float(selectedContent.voteAverage*0.1)
         contentTitle.text = selectedContent.title
         contentOverview.text = selectedContent.overview
-        releaseDate.text = selectedContent.release_date
-        
-        let url = URL(string: "https://image.tmdb.org/t/p/w500/\(selectedContent.poster_path)")
+        releaseDate.text = selectedContent.releaseDate
+        let url = URL(string: "https://image.tmdb.org/t/p/w500/\(selectedContent.posterPath)")
```
Comment (cordhernandez) [inline Marquee/ViewControllers/DetailViewController.swift:33]: Might be a good idea to keep all of your raw strings in a Constant struct. I left a similar comment in one of your other PRs.

File: Marquee/ViewControllers/FavoritesViewController.swift [hunk 7]
```diff
@@ -54,23 +47,21 @@ class FavoritesViewController: UIViewController, UICollectionViewDelegate, UICol
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         let cell = favoritesCollectionView.dequeueReusableCell(withReuseIdentifier: CollectionViewCell.identifier, for: indexPath) as! CollectionViewCell
-        
-        let url = URL(string: "https://image.tmdb.org/t/p/w500/\(FavoritesList.shared.list[indexPath.row].poster_path!)")
+        let url = URL(string: "https://image.tmdb.org/t/p/w500/\(FavoritesList.shared.list[indexPath.row].posterPath!)")
```
Comment (cordhernandez) [inline Marquee/ViewControllers/FavoritesViewController.swift]: careful with the force unwrapping here.. it's not a good look ;)

File: Marquee/ViewControllers/FavoritesViewController.swift [hunk 8]
```diff
@@ -54,23 +47,21 @@ class FavoritesViewController: UIViewController, UICollectionViewDelegate, UICol
...
         return 0
     }
     
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        
         let selectedItem = Content(overview: FavoritesList.shared.list[indexPath.row].overview!,
-                                   poster_path: FavoritesList.shared.list[indexPath.row].poster_path!,
-                                   release_date: FavoritesList.shared.list[indexPath.row].release_date!,
+                                   posterPath: FavoritesList.shared.list[indexPath.row].posterPath!,
+                                   releaseDate: FavoritesList.shared.list[indexPath.row].releaseDate!,
```
Comment (cordhernandez) [inline Marquee/ViewControllers/FavoritesViewController.swift]: same here, careful with the force unwrapping.

File: Marquee/ViewControllers/SearchViewController.swift [hunk 9]
```diff
@@ -14,15 +14,14 @@ class SearchViewController: UIViewController {
...
     let jsonManager = JSONManager()
     var searchData: [SearchObject]? {
         didSet {
             DispatchQueue.main.async {
                 self.searchCollectionView.reloadData()
             }
-            
-        }}
+        }
+    }
```
Comment (cordhernandez) [inline Marquee/ViewControllers/SearchViewController.swift:24]: nice clean up 👏🏾

File: Marquee/ViewControllers/SearchViewController.swift [hunk 10]
```diff
@@ -74,22 +68,21 @@ extension SearchViewController: UICollectionViewDelegate, UICollectionViewDataSo
...
+        let releaseDate = searchData?[indexPath.row].firstAirDate ?? searchData?[indexPath.row].releaseDate
         
         let selectedItem = Content(overview: (searchData?[indexPath.row].overview)!,
-                                   poster_path: (searchData?[indexPath.row].poster_path)!,
-                                   release_date: releaseDate!,
+                                   posterPath: (searchData?[indexPath.row].posterPath)!,
+                                   releaseDate: releaseDate!,
                                    title: title!,
-                                   vote_average: (searchData?[indexPath.row].vote_average)!)
+                                   voteAverage: (searchData?[indexPath.row].voteAverage)!)
```
Comment (cordhernandez) [inline Marquee/ViewControllers/SearchViewController.swift]: again, careful with the force unwrap.

File: Marquee/ViewControllers/ViewController.swift [hunk 11]
```diff
@@ -177,14 +157,15 @@ extension ViewController:UITableViewDelegate, UITableViewDataSource, collectionV
...
             newFavorite.title = selectedContent.title
             newFavorite.overview = selectedContent.overview
-            newFavorite.poster_path = selectedContent.poster_path
-            newFavorite.release_date = selectedContent.release_date
-            newFavorite.vote_average = selectedContent.vote_average
-            
+            newFavorite.posterPath = selectedContent.posterPath
+            newFavorite.releaseDate = selectedContent.releaseDate
+            newFavorite.voteAverage = selectedContent.voteAverage
+            print(newFavorite)
```
Comment (cordhernandez) [inline Marquee/ViewControllers/ViewController.swift:169]: Print statements are a poor mans way of logging lol ... might want to consider removing print statement for production code and just keep them for debugging. Also using Apple's built in logging system might not be a bad idea.

**Review by cordhernandez — COMMENTED** [review]
@JoshuaBayless Not bad not bad! The code is looking a lot cleaner :) 

I did leave quite a few comments, take a look when you can.

**Comment by JoshuaBayless** [comment]
Removed all force unwrapping and replaced with optional chaining.

File: Marquee/Cells/Table Cell/TableViewCell.swift (hunk 1 above)
Comment (JoshuaBayless) [reply Marquee/Cells/Table Cell/TableViewCell.swift:19]: I believe my thinking there was that I didn't want to be limited to a UIViewController if in the future I decided to use a UICollectionViewController or UITableViewController. 

Would this be considered bad practice?

**Pull request merged**